pkg/operator/staticpod/controller/installer/installer_controller_test.go

```diff
@@ -884,6 +885,37 @@ func TestCreateInstallerPodMultiNode(t *testing.T) {
 			updateStatusErrors:   []error{errors.NewInternalError(fmt.Errorf("unknown"))},
 			expectedSyncError:    []bool{true},
 		},
+		{
+			name:                    "three nodes, 2 not updated, one with failure in last revision which is no longer available",
+			latestAvailableRevision: 3,
+			nodeStatuses: []operatorv1.NodeStatus{
+				{
+					NodeName:        "test-node-0",
+					CurrentRevision: 1,
+				},
+				{
+					NodeName:           "test-node-1",
+					CurrentRevision:    1,
+					LastFailedRevision: 2,
+				},
+				{
+					NodeName:        "test-node-2",
+					CurrentRevision: 1,
+				},
+			},
+			staticPods: []*corev1.Pod{
+				newStaticPod(mirrorPodNameForNode("test-pod", "test-node-1"), 1, corev1.PodRunning, true),
+				newStaticPod(mirrorPodNameForNode("test-pod", "test-node-2"), 2, corev1.PodRunning, true),
+				newStaticPod(mirrorPodNameForNode("test-pod", "test-node-3"), 1, corev1.PodRunning, true),
+			},
+			expectedUpgradeOrder: []int{0, 2, 1},
```

Comment:
Do we actually test the case of "target version set, target version < latestAvailableRevision, target version failing" case?

What does 0,2,1 mean in your case?

Reply:
The order in which the pods are created. I've change the test such that no pods exist at all (after all, they were deleted), and updated the order to match.